Tests/test_SffIO.py

```diff
@@ -368,181 +368,104 @@ def test_index2(self):
-        fasta_no_trim = list(SeqIO.parse(filename, "fasta"))
-        filename = "Roche/E3MFGYR02_random_10_reads_no_trim.qual"
-        qual_no_trim = list(SeqIO.parse(filename, "qual"))
-
-        filename = "Roche/E3MFGYR02_random_10_reads.fasta"
-        fasta_trim = list(SeqIO.parse(filename, "fasta"))
-        filename = "Roche/E3MFGYR02_random_10_reads.qual"
-        qual_trim = list(SeqIO.parse(filename, "qual"))
-
-        for s, sT, f, q, fT, qT in zip(sff, sff_trim, fasta_no_trim, qual_no_trim, fasta_trim, qual_trim):
-            assert s.id == f.id == q.id
-            assert str(s.seq) == str(f.seq)
-            assert s.letter_annotations["phred_quality"] == q.letter_annotations["phred_quality"]
-            assert s.id == sT.id == fT.id == qT.id
-            assert str(sT.seq) == str(fT.seq)
-            assert sT.letter_annotations["phred_quality"] == qT.letter_annotations["phred_quality"]
+        with open(filename, "rb") as a_handle, open(filename, "rb") as b_handle:
+            sff = list(SffIterator(a_handle))
+            sff_trim = list(SffIterator(b_handle, trim=True))
+
+            filename = "Roche/E3MFGYR02_random_10_reads_no_trim.fasta"
+            fasta_no_trim = list(SeqIO.parse(filename, "fasta"))
+            filename = "Roche/E3MFGYR02_random_10_reads_no_trim.qual"
+            qual_no_trim = list(SeqIO.parse(filename, "qual"))
+
+            filename = "Roche/E3MFGYR02_random_10_reads.fasta"
+            fasta_trim = list(SeqIO.parse(filename, "fasta"))
+            filename = "Roche/E3MFGYR02_random_10_reads.qual"
+            qual_trim = list(SeqIO.parse(filename, "qual"))
+
+            for s, sT, f, q, fT, qT in zip(sff, sff_trim, fasta_no_trim, qual_no_trim, fasta_trim, qual_trim):
+                self.assertEqual(len({s.id, f.id, q.id}), 1)  # All values are the same
+                self.assertEqual(str(s.seq), str(f.seq))
+                self.assertEqual(s.letter_annotations["phred_quality"], q.letter_annotations["phred_quality"])
+                self.assertEqual(len({s.id, sT.id, fT.id, qT.id}), 1)  # All values are the same
+                self.assertEqual(str(sT.seq), str(fT.seq))
+                self.assertEqual(sT.letter_annotations["phred_quality"], qT.letter_annotations["phred_quality"])
 
     def test_write(self):
         filename = "Roche/E3MFGYR02_random_10_reads.sff"
-        with open(filename, "rb") as handle:
-            metadata = ReadRocheXmlManifest(handle)
-        with open(filename, "rb") as handle:
-            sff = list(SffIterator(handle))
-        handle = BytesIO()
-        w = SffWriter(handle, xml=metadata)
-        w.write_file(sff)  # list
-        data = handle.getvalue()
-        # And again with an iterator...
-        handle = BytesIO()
-        w = SffWriter(handle, xml=metadata)
-        w.write_file(iter(sff))
-        assert data == handle.getvalue()
+        with open(filename, "rb") as a_handle, open(filename, "rb") as b_handle:
```

Comment:
Again, same comment as before - the with clauses should be as small as possible - as per the original.